src/components/Store.ts

```diff
@@ -24,44 +35,115 @@ export class Store {
-            [key : string] : IInstance;
-        }>(InfluxDBInstancesKey) || {}
+    // Get the config. Don't cache this, otherwise we'd have to set up change
+    // event listeners.
+    get config() : vscode.WorkspaceConfiguration {
+        return vscode.workspace.getConfiguration('vsflux')
+    }
+
+    async getInstances() : Promise<{ [key : string] : IInstance }> {
+        const dataSource = this.config.get<string>('datasource', 'db')
+        if (dataSource === 'cli') {
+            const { stdout, stderr } = await exec('influx config list --json')
+            if (stderr) {
+                console.error(`Error from influx cli: ${stdout}`)
+                vscode.window.showErrorMessage('Error invoking the influx command')
+                return {}
+            }
+
+            const cliData : { [key : string] : CLIInstance } = JSON.parse(stdout)
+            const instances : { [key : string] : IInstance } = {}
+            for (const [id, config] of Object.entries(cliData)) {
+                instances[id] = {
+                    id,
+                    disableTLS: false,
+                    hostNport: config.url,
+                    name: id,
+                    org: config.org,
+                    token: config.token,
+                    isActive: config.active !== undefined ? config.active : false
+                }
+            }
+            return instances
+        } else {
+            return this.context.globalState.get<{
+                [key : string] : IInstance;
+            }>(InfluxDBInstancesKey) || {}
+        }
     }
 
-    getInstance(id : string) : IInstance {
-        return this.getInstances()[id]
+    async getInstance(id : string) : Promise<IInstance> {
+        const instances = await this.getInstances()
+        return instances[id]
     }
 
     async saveInstance(connection : IInstance) : Promise<void> {
-        const connections = this.getInstances()
-        if (connection.isActive) {
-            // Ensure no other connection is marked active
-            for (const [key, _] of Object.entries(connections)) {
-                if (key !== connection.id) {
-                    connections[key].isActive = false
+        const connections = await this.getInstances()
+        const dataSource = await this.config.get<string>('datasource', 'db')
+        if (dataSource === 'cli') {
+            if (connections[connection.id] === undefined) { // New instance
+                const createCommand = `influx config create \
+                    --config-name ${connection.name} \
+                    --host-url ${connection.hostNport} \
+                    --token ${connection.token} \
+                    --org ${connection.org}`
+                const { stdout, stderr } = await exec(createCommand)
+                if (stderr) {
+                    console.error(`Error from influx cli: ${stdout}`)
+                    vscode.window.showErrorMessage('Error invoking the influx command')
```

Comment:
Maybe display the cli error here for the user?

Reply:
I'm not sure what the error is going to be, so I'd prefer not to display the error in the error message. Our users have shown they're not afraid to crack open the console if need be.

Reply:
Might just be me, but I despise errors omit relevant information. Better that the error information is strangely formatted/worded than the information being hidden away, forcing me to jump through hoops to actually fix it.

Reply:
Yeah, I think there's a balancing act here. You can either say "an error occurred" and tell the user where to find more information, _or_ put a potentially truncated error in a small pop up on the bottom right, which will have awkward copy/paste dynamics. I don't think that `showErrorMessage` UI was intended to show the kinds of error messages that we might need to show, which is why I opted to not use it.